res/controllers/Numark-Mixtrack-Platinum-FX-scripts.js

```diff
@@ -0,0 +1,1899 @@
+		MixtrackPlatinumFX.sendScreenBpmMidi(i+1,0);
+		MixtrackPlatinumFX.sendScreenTimeMidi(i+1,0);
+		MixtrackPlatinumFX.sendScreenDurationMidi(i+1,0);		
+	}
+
+	// switch to decks 1 and 2
+	midi.sendShortMsg(0x90, 0x08, 0x7F);
+	midi.sendShortMsg(0x91, 0x08, 0x7F);
+	
+    midi.sendSysexMsg(shutdownSysex, shutdownSysex.length);
+};
+
+MixtrackPlatinumFX.shift = function() {
+    MixtrackPlatinumFX.shifted = true;
+    MixtrackPlatinumFX.deck.shift();
+    MixtrackPlatinumFX.browse.shift();
+    MixtrackPlatinumFX.effect.shift();
+	MixtrackPlatinumFX.gains.cueGain.shift();
+};
+
+MixtrackPlatinumFX.unshift = function() {
+    MixtrackPlatinumFX.shifted = false;
+    MixtrackPlatinumFX.deck.unshift();
+    MixtrackPlatinumFX.browse.unshift();
+    MixtrackPlatinumFX.effect.unshift();
+	MixtrackPlatinumFX.gains.cueGain.unshift();
+};
+
+MixtrackPlatinumFX.allEffectOff = function() {
+	MixtrackPlatinumFX.effect[0].effects=[false, false, false];
+	MixtrackPlatinumFX.effect[1].effects=[false, false, false];
+	MixtrackPlatinumFX.FxBlinkUpdateLEDs();
+	MixtrackPlatinumFX.effect[0].updateEffects();
+	MixtrackPlatinumFX.effect[1].updateEffects();
+};
+
+MixtrackPlatinumFX.FxBlinkUpdateLEDs = function() {
+	var newStates1=[false, false, false];
+	var newStates2=[false, false, false];
+	if (!MixtrackPlatinumFX.FxBlinkState || MixtrackPlatinumFX.BlinkState) {
+		newStates1=MixtrackPlatinumFX.effect[0].effects;
+		newStates2=MixtrackPlatinumFX.effect[1].effects;
+	}
+    midi.sendShortMsg(0x98, 0x00, newStates1[0] ? MixtrackPlatinumFX.HIGH_LIGHT:MixtrackPlatinumFX.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x01, newStates1[1] ? MixtrackPlatinumFX.HIGH_LIGHT:MixtrackPlatinumFX.LOW_LIGHT);
+    midi.sendShortMsg(0x98, 0x02, newStates1[2] ? MixtrackPlatinumFX.HIGH_LIGHT:MixtrackPlatinumFX.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x03, newStates2[0] ? MixtrackPlatinumFX.HIGH_LIGHT:MixtrackPlatinumFX.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x04, newStates2[1] ? MixtrackPlatinumFX.HIGH_LIGHT:MixtrackPlatinumFX.LOW_LIGHT);
+    midi.sendShortMsg(0x99, 0x05, newStates2[2] ? MixtrackPlatinumFX.HIGH_LIGHT:MixtrackPlatinumFX.LOW_LIGHT);
+}
+
+MixtrackPlatinumFX.FxBlinkTimer=0;
+MixtrackPlatinumFX.FxBlinkState=true;
+MixtrackPlatinumFX.FxBlink = function() {
+	var start = MixtrackPlatinumFX.effect[0].isSwitchHolded || MixtrackPlatinumFX.effect[1].isSwitchHolded;
+	
+	if (start) {
+		MixtrackPlatinumFX.FxBlinkState = true;
+		MixtrackPlatinumFX.FxBlinkUpdateLEDs();
+	} else {
+		// stop
+        MixtrackPlatinumFX.FxBlinkState = false;
+		MixtrackPlatinumFX.FxBlinkUpdateLEDs();
+	}
+};
+
+// TODO in 2.3 it is not possible to "properly" map the FX selection buttons.
+// this should be done with load_preset and QuickEffects instead (when effect
+// chain preset saving/loading is available in Mixxx)
+MixtrackPlatinumFX.EffectUnit = function(deckNumber) {    
+    this.effects = [false, false, false];
+    this.isSwitchHolded = false;
+        
```

Comment:
this todo and implementation is now obsolete since you're targetting mixxx 2.4 and `loaded_chain_preset` is available (just not yet documented, PR is in progress though, see the preview below).

https://deploy-preview-612--mixxx-manual.netlify.app/chapters/appendix/mixxx_controls#control-[QuickEffectRack1_[ChannelI]]-loaded_chain_preset

Reply:
WIll check on this..